### DISCUSSION TITLE
v0.4.1

### DISCUSSION THREAD

**Discussion by mattdangerw** [keras-team/keras-hub#775] [General]
The 0.4.1 release is a minor release with new model architectures and compilation defaults for task models. If you encounter any problems or have questions, please open an issue!

## Summary
* Added compilation defaults for all **task** models (e.g. `keras_nlp.models.BertClassifier`). No existing functionality is changed, but users of task models can now skip calling `.compile()` and use default learning rates and optimizer strategies provided by the library.
* Added `keras_nlp.models.AlbertBackbone`, `keras_nlp.models.AlbertClassifier`, preprocessor, and tokenizer layers for pre-trained [ALBERT](https://arxiv.org/abs/1909.11942) models.
* Added `keras_nlp.models.FNetBackbone`, `keras_nlp.models.FNetClassifier`, preprocessor, and tokenizer layers for pre-trained [FNet](https://arxiv.org/abs/2105.03824) models.
* Added `keras_nlp.models.DebertaV3Backbone`, `keras_nlp.models.DebertaV3Classifier`, preprocessor, and tokenizer layers for pre-trained [DeBERTaV3](https://arxiv.org/abs/2111.09543) models.

## What's Changed
* Update python version in readme to 3.8 by @haifeng-jin in https://github.com/keras-team/keras-nlp/pull/618
* Modify our pip install line so we upgrade tf by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/616
* Use Adam optimizer for quick start by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/620
* Clean up class name and `self` in calls to `super()` by @mbrukman in https://github.com/keras-team/keras-nlp/pull/628
* Update word_piece_tokenizer.py by @ADITYADAS1999 in https://github.com/keras-team/keras-nlp/pull/617
* Add DeBERTaV3 Conversion Script by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/633
* Add AlbertTokenizer and AlbertPreprocessor by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/627
* Create `Backbone` base class by @jbischof in https://github.com/keras-team/keras-nlp/pull/621
* Add TPU testing by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/591
* Add Base Preprocessor Class by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/638
* Add keras_nlp.samplers by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/563
* Add ALBERT Backbone by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/622
* Add a small script to count parameters in our presets by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/610
* Clean up examples/ directory by @ADITYADAS1999 in https://github.com/keras-team/keras-nlp/pull/637
* Fix Small BERT Typo by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/651
* Rename examples/bert -> examples/bert_pretraining by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/647
* Add FNet Preprocessor by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/646
* Add FNet Backbone by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/643
* Small DeBERTa Docstring Fixes by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/666
* Add Fenced Docstring Testing by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/640
* Corrected the epsilon value  by @soma2000-lang in https://github.com/keras-team/keras-nlp/pull/665
* Consolidate docstring formatting weirdness in Backbone and Preprocessor base classes by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/654
* Fix  `value_dim` in `TransformerDecoder`'s cross-attn layer by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/667
* Add ALBERT Presets by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/655
* Add Base Task Class by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/671
* Implement TopP, TopK and Beam samplers by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/652
* Add FNet Presets by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/659
* Bump the year to 2023 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/679
* Add BART Backbone by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/661
* Handle trainable and name in the backbone base class by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/680
* Ignore Task Docstring for Testing by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/683
* Light-weight benchmarking script by @NusretOzates in https://github.com/keras-team/keras-nlp/pull/664
* Conditionally import tf_text everywhere by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/684
* Expose `token_embedding` as a Backbone Property by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/676
* Move `from_preset` to base tokenizer classes by @shivance in https://github.com/keras-team/keras-nlp/pull/673
* add f_net_classifier and f_net_classifier_test by @ADITYADAS1999 in https://github.com/keras-team/keras-nlp/pull/670
* import rouge_scorer directly from rouge_score package by @sampathweb in https://github.com/keras-team/keras-nlp/pull/691
* Fix typo in requirements file juypter -> jupyter by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/693
* Temporary fix to get nightly green again by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/696
* GPT2 Text Generation APIs by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/592
* Run keras saving tests on nightly and fix RobertaClassifier test by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/692
* Speed up pip install keras-nlp; simplify deps by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/697
* Add `AlbertClassifier` by @shivance in https://github.com/keras-team/keras-nlp/pull/668
* Make tokenizer, backbone, preprocessor properties settable on base class by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/700
* Update to latest black by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/708
* RobertaMaskedLM task and preprocessor by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/653
* Default compilation for BERT/RoBERTa classifiers by @jbischof in https://github.com/keras-team/keras-nlp/pull/695
* Add start/end token padding to `GPT2Preprocessor` by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/704
* Don't install tf stable when building our nightly image by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/711
* Add OPT Backbone and Tokenizer by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/699
* Small OPT Doc-string Edits by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/716
* Default compilation other classifiers by @Plutone11011 in https://github.com/keras-team/keras-nlp/pull/714
* Add BartTokenizer and BART Presets by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/685
* Add an add_prefix_space Arg in BytePairTokenizer by @shivance in https://github.com/keras-team/keras-nlp/pull/715
* Opt presets by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/707
* fix import of tensorflow_text in tf_utils by @sampathweb in https://github.com/keras-team/keras-nlp/pull/723
* Check for masked token in roberta tokenizer by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/742
* Improve test coverage for special tokens in model tokenizers by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/743
* Fix the sampler truncation strategy by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/713
* Add ALBERT Conversion Script by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/736
* Add FNet Conversion Script by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/737
* Add BART Conversion Script by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/739
* Pass Correct LayerNorm Epsilon value to TransformerEncoder in Backbones by @TheAthleticCoder in https://github.com/keras-team/keras-nlp/pull/731
* Improving the layer Description. by @Neeshamraghav012 in https://github.com/keras-team/keras-nlp/pull/734
* Adding ragged support to SinePositionEncoding by @apupneja in https://github.com/keras-team/keras-nlp/pull/751
* Fix trailing space by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/755
* Adding an AlbertMaskedLM task model and preprocessor by @shivance in https://github.com/keras-team/keras-nlp/pull/725
* New docstring example for TokenAndPosition Embedding layer. by @Neeshamraghav012 in https://github.com/keras-team/keras-nlp/pull/760
* Add a note for TPU issues for deberta_v3 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/758
* Add missing exports to models API by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/763
* Autogenerate preset table by @Cyber-Machine in https://github.com/keras-team/keras-nlp/pull/690
* Remove work in progress API for 0.4.1 release by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/765
* Version bump to 0.4.1.dev0 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/766
* Version bump to 0.4.1 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/768

## New Contributors
* @haifeng-jin made their first contribution in https://github.com/keras-team/keras-nlp/pull/618
* @mbrukman made their first contribution in https://github.com/keras-team/keras-nlp/pull/628
* @soma2000-lang made their first contribution in https://github.com/keras-team/keras-nlp/pull/665
* @NusretOzates made their first contribution in https://github.com/keras-team/keras-nlp/pull/664
* @shivance made their first contribution in https://github.com/keras-team/keras-nlp/pull/673
* @Plutone11011 made their first contribution in https://github.com/keras-team/keras-nlp/pull/714
* @TheAthleticCoder made their first contribution in https://github.com/keras-team/keras-nlp/pull/731
* @Neeshamraghav012 made their first contribution in https://github.com/keras-team/keras-nlp/pull/734
* @Cyber-Machine made their first contribution in https://github.com/keras-team/keras-nlp/pull/690

**Full Changelog**: https://github.com/keras-team/keras-nlp/compare/v0.4.0...v0.4.1

<hr /><em>This discussion was created from the release <a href='https://github.com/keras-team/keras-nlp/releases/tag/v0.4.1'>v0.4.1</a>.</em>

**Comment by athleticcoder21** [comment]
@mattdangerw We also fixed Mistake in projection layer dimension of MaskedLMHead #733 through #725

**Comment by mattdangerw** [comment]
@shivance indeed we did! Though actually we are not exposing the new `XXMaskedLM` task models quite yet (because they are still in the process of landing for all models). Overall that is something we will highlight more for the upcoming 0.5 release.

The fix for `MaskedLMHead` head is live on 0.4.1 though and much appreciated!
